File: programmers/level2/87390.js
//https://school.programmers.co.kr/learn/courses/30/lessons/87390
//n^2 배열 자르기

function solution(n, left, right) {
  const answer = [];

  for (let i = 0; i < n; i++) {
    // console.log('i+1',i+1)
    for (let j = 1; j <= n; j++) {
      // console.log('j',j)
      if (j < i + 1) {
        answer.push(i + 1);
      } else {
        answer.push(j);
      }
    }
  }

  return answer.slice(left, right + 1);
}

// console.log(solution(3, 2, 5)); //[3,2,2,3]
console.log(solution(4, 7, 14)); //[4,3,3,3,4,4,4,4]

File: programmers/python/level2/87390.py
def solution(n, left, right):
  answer = []
  for idx in range(left, right+1):
    row, col = divmod(idx,n)
    answer.append(max(row, col)+1)
  return answer

print(solution(3,	2,	5))
# print(solution(4,	7,	14))
# 3	2	5	[3,2,2,3]
# 4	7	14	[4,3,3,3,4,4,4,4]
